src/main/java/com/github/mizool/jediscache/JedisCacheManager.java

```diff
@@ -0,0 +1,219 @@
+    {
+        verifyOpen();
+        Cache<K, V> cache;
+        synchronized (caches)
+        {
+            cache = getCacheInstance(cacheName);
+        }
+        if (cache != null)
+        {
+            Configuration<K, V> configuration = cache.getConfiguration(CompleteConfiguration.class);
+
+            if (configuration.getKeyType() == null || !configuration.getKeyType().equals(keyType))
+            {
+                throw new ClassCastException("Incompatible cache key types specified, expected " +
+                    configuration.getKeyType() +
+                    " but " +
+                    keyType +
+                    " was specified");
+            }
+
+            if (configuration.getValueType() == null || !configuration.getValueType().equals(valueType))
+            {
+                throw new ClassCastException("Incompatible cache value types specified, expected " +
+                    configuration.getValueType() +
+                    " but " +
+                    valueType +
+                    " was specified");
+            }
+        }
+
+        return cache;
+    }
+
+    @Override
+    public <K, V> Cache<K, V> getCache(String cacheName)
+    {
+        synchronized (caches)
+        {
+            verifyOpen();
+            return getCacheInstance(cacheName);
+        }
+    }
+
+    @Override
+    public Iterable<String> getCacheNames()
+    {
+        synchronized (caches)
+        {
+            verifyOpen();
+            return ImmutableSet.copyOf(caches.keySet());
+        }
+    }
+
+    @Override
+    public void destroyCache(@NonNull String cacheName)
+    {
+        synchronized (caches)
+        {
+            verifyOpen();
+            Cache<?, ?> cache = caches.get(cacheName);
+            if (cache != null)
+            {
+                cache.close();
+            }
+        }
+    }
+
+    @Override
+    public void enableManagement(@NonNull String cacheName, boolean enabled)
+    {
+        throw new UnsupportedOperationException("JMX features are not supported");
+    }
+
+    @Override
+    public void enableStatistics(@NonNull String cacheName, boolean enabled)
+    {
+        throw new UnsupportedOperationException("JMX features are not supported");
+    }
+
+    @Override
+    public void close()
+    {
+        if (!closed.getAndSet(true))
+        {
+            cachingProvider.releaseCacheManager(getURI(), getClassLoader());
+
+            Collection<Cache<?, ?>> cachesToClose;
+            synchronized (caches)
+            {
+                cachesToClose = Lists.newArrayList(caches.values());
+                caches.clear();
+            }
+            for (Cache<?, ?> cache : cachesToClose)
+            {
+                try
+                {
+                    cache.close();
+                }
+                catch (RuntimeException e)
+                {
+                    log.warn("Error stopping cache: " + cache, e);
```

Comment:
for this to be useful, we need to write a `toString()` implementation in `JedisCache` - the one generated by `@Data` includes a bunch of non-helpful fields (e.g. the cache manager). we should limit this to the cache name (probably lombok can do this for us).